File: code_samples/lua/database_migrations.lua
-- Database: Migrations
-- AI/ML Training Sample

Migrations = {}
Migrations.__ index = Migrations

function Migrations:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Migrations:process(input)
    self.data = input
end

function Migrations:getData()
    return self.data
end

function Migrations:validate()
    return #self.data > 0
end

-- Example usage
local instance = Migrations:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/dart/database_migrations.dart
// Database: Migrations
// AI/ML Training Sample

class Migrations {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = Migrations();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
